db/external_sst_file_ingestion_job.cc

```diff
@@ -160,9 +160,9 @@ Status ExternalSstFileIngestionJob::Prepare(
       TEST_SYNC_POINT_CALLBACK("ExternalSstFileIngestionJob::Prepare:CopyFile",
                                nullptr);
       // CopyFile also sync the new file.
-      status =
-          CopyFile(fs_.get(), path_outside_db, path_inside_db, 0,
-                   db_options_.use_fsync, io_tracer_, Temperature::kUnknown);
+      status = CopyFile(fs_.get(), path_outside_db, path_inside_db, 0,
```

Comment:
The external file ingestion job supports a per ingestion temperature option: https://github.com/facebook/rocksdb/blob/a43481b3d06f132e0b4d0d9d272d59eedbf09065/include/rocksdb/db.h#L134

This option seems to be missed from this `CopyFile` invocation but was used in some other places like creating `FileMetaData`. Shall we consolidate the priority of these two options: per ingestion file temperature, per column family default write file temperature. And use the chosen one consistently in all places?

Reply:
There's quite a bit to unravel and fix with temperatures and ingestion, so I'm going to defer that to a follow-up PR, and put a hefty FIXME here in this PR.